Part 2 of Joe TNS post (#3594)
pulumi/pulumi-hugo: themes/default/content/blog/walkthrough-of-adopting-infrastructure-as-code/index.md

---
title: "A Walkthrough of Adopting Infrastructure as Code"
canonical_url: https://thenewstack.io/a-walkthrough-of-adopting-infrastructure-as-code/

> date: 2023-11-07T19:54:55Z

Author: ```suggestion
date: 2023-11-14T19:54:55Z
```

draft: false
meta_desc: A look at the basics of how to get going with Infrastructure as Code, and then a dive into some advanced use cases to show what you can do from there.
meta_image: meta.png
authors:
    - joe-duffy
tags:
    - infrastructure-as-code
    - fundamentals

---

*This is the second of a three-part series originally published on The New Stack.*

> Following the first piece in this series, [Infrastructure as Code in Any Programming Language](/blog/infrastructure-as-code-in-any-programming-language/), this walkthrough will show what it takes to get up and running with Infrastructure as Code. Everything we show will be done with Pulumi’s free and open source Infrastructure as Code SDK, which you can [download here](/docs/install/). You can also sign up for Pulumi Cloud [here](https://app.pulumi.com/signup). After discussing the basics of how to get going, we’ll then dive into some advanced use cases to show what you can do from there.

Reviewer: I'm assuming this is intended to be a faithful repost of the original TNS article (i.e., no content changes). If not, I'd recommend rewording some of the links to avoid the use of "here" in the hyperlinks.

Author: I think we can make some simple edits. What do you suggest we say instead of "here"?

Reviewer: ```suggestion
Following the first piece in this series, [Infrastructure as Code in Any Programming Language](/blog/infrastructure-as-code-in-any-programming-language/), this walkthrough will show what it takes to get up and running with Infrastructure as Code. Everything we show will be done with Pulumi’s [free and open source Infrastructure as Code SDK](/docs/install/). You can also sign up for [Pulumi Cloud](https://app.pulumi.com/signup). After discussing the basics of how to get going, we’ll then dive into some advanced use cases to show what you can do from there.
```

![AWS Kubernetes Template](/templates/kubernetes/aws/architecture.png)

*See the template for AWS [here](/templates/kubernetes/aws/)*.

You can see here even greater differences between the equivalent cloud services on different cloud providers and AWS’s more “building blocks” style of approach in action. This is also a great example of Infrastructure as Code’s ability to bundle best practices so that they can be reused. This uses the [Pulumi EKS Package](/registry/packages/eks/), which configures all of the above in the standard way so that you don’t need to figure out every little detail when you’re just getting started.

In both cases, we end up with a fully managed Kubernetes cluster that is ready to use. The tool spits out a kubeconfig file so we can immediately deploy into or inspect it.

## Infrastructure Projects and Stacks

All of these examples had one thing in common: The Infrastructure as Code program’s job is to declare individual resource objects, their properties and how they relate to each other. That was true no matter the scenario, the cloud or the language we have chosen. Once the program has done that, the Infrastructure as Code tool takes care of the rest.

> The code that declares all of this is something Pulumi calls a **project**, and each instance of said project that is configured and ready to deploy — each “environment,” if you will — is called a **stack**.

Reviewer: ```suggestion
Pulumi uses the term **project** to refer to the code that declares all of this, and each instance of said project that is configured and ready to deploy — each “environment,” if you will — is called a **stack**.
```
